### PR TITLE
add mocking and basic test for storage pods

### PR DESCRIPTION
**Pull request by timogoebel**
This PR adds some basic mocking and a basic test... Nothing fancy, but a start...

### CODE REVIEW

File: lib/fog/vsphere/requests/compute/get_storage_pod.rb [hunk 1]
```diff
@@ -23,6 +23,7 @@ def get_raw_storage_pod(name, datacenter_name)
 
       class Mock
         def get_storage_pod(name, datacenter_name)
+          list_storage_pods().select{|h| h[:name] == name }.first
```
Comment (plribeiro3000) [inline lib/fog/vsphere/requests/compute/get_storage_pod.rb]: I believe the mock version of the api should be more like the real one. Can't you do a search based on the `datacenter_name`?

**Comment by plribeiro3000** [comment]
@timogoebel Looks awesome. Just a small question that i believe there is nothing to do. I just want to make sure i'm not missing it.

**Comment by timogoebel** [comment]
@plribeiro3000 : Improved this a bit, mimics the real behavior more closely.

**Comment by plribeiro3000** [comment]
It looks awesome. Thank you for taking the time to improve it!

**Comment by plribeiro3000** [comment]
Thanks!

**Pull request merged**